boards/pyportal/src/pins.rs

```diff
@@ -1,116 +1,350 @@
-    /// TFT RD
-    pin tft_rd = b4,
-    /// TFT RS
-    pin tft_rs = b5,
-    /// TFT CS
-    pin tft_cs = b6,
-    /// TFT TE
-    pin tft_te = b7,
-    /// TFT WR
-    pin tft_wr = b9,
-    /// TFT Backlight
-    pin tft_backlight = b31,
-
-    /// LCD Data 0
-    pin lcd_data0 = a16,
-    /// LCD Data 1
-    pin lcd_data1 = a17,
-    /// LCD Data 2
-    pin lcd_data2 = a18,
-    /// LCD Data 3
-    pin lcd_data3 = a19,
-    /// LCD Data 4
-    pin lcd_data4 = a20,
-    /// LCD Data 5
-    pin lcd_data5 = a21,
-    /// LCD Data 6
-    pin lcd_data6 = a22,
-    /// LCD Data 7
-    pin lcd_data7 = a23,
-
-    /// Touchscreen pins
-    /// Touch YD
-    pin touch_yd = b0,
-    /// Touch XL
-    pin touch_xl = b1,
-    /// Touch YU
-    pin touch_yu = a6,
-    /// Touch XR
-    pin touch_xr = b8,
-
-    // ESP control - ESP32 WiFi
-    /// Pin ESP CS
-    pin esp_cs = b14,
-    /// Pin ESP GPIO0
-    pin esp_gpio0 = b15,
-    /// Pin ESP Busy
-    pin esp_busy = b16,
-    /// Pin ESP Reset
-    pin esp_reset = b17,
-    /// Pin ESP RTS
-    pin esp_rts = a15,
-
-    // UART - Universal Asynchronous Receiver/Transmitter (connected to ESP32)
-    /// Pin TX
-    pin esp_tx = b12,
-    /// Pin RX
-    pin esp_rx = b13,
-
-    // SPI - Serial Peripheral Interface
-    /// Pin MOSI
-    pin mosi = a12,
-    /// Pin SCK
-    pin sck = a13,
-    /// Pin MISO
-    pin miso = a14,
-
-    // I2C - ADT7410 Analog Devices temperature sensor
-    /// Pin SDA
-    pin sda = b2,
-    /// Pin SCL
-    pin scl = b3,
-
-    /// Pin SD CS
-    pin sd_cs = b30,
-    /// Pin SD card detect
-    pin sd_card_detect = a1,
+use super::hal;
+
+hal::bsp_pins!(
+    PA02 {
+        /// Analog pin 0.  Can act as a true analog output
+        /// as it has a DAC (which is not currently supported
+        /// by this hal) as well as input.
+        name: speaker,
+        aliases: {
+            Reset: SpeakerReset,
+        }
+    }
+    PA27 {
+        /// enable speaker amplifier
+        name: speaker_enable,
+        aliases: {
+            Reset: SpeakerEnableReset,
+        }
+    }
+    PA07 {
+        /// Light sensor
+        name: light,
+        aliases: {
+            AlternateB: LightSensorAdc,
+            Reset: LightSensorReset,
+        }
+    }
+    PA04 {
+        // STEMMA connectors
+        /// Pin D3
+        name: d3,
+        aliases: {
+            Reset: D3Reset,
+        }
+    }
+    PA05 {
+        /// Pin D4
+        name: d4,
+        aliases: {
+            Reset: D4Reset,
+        }
+    }
+    PB23 {
+        /// D13 LED
+        name: d13,
+        aliases: {
+            PushPullOutput: RedLed,
+            Reset: RedLedReset,
+        }
+    }
+    PB22 {
+        /// Neopixel status LED
+        name: neopixel,
+        aliases: {
+            Reset: NeopixelReset,
+            PushPullOutput: Neopixel,
+        }
+    }
+
+    PA00 {
+        // TFT(Thin-film-transistor liquid-crystal display) control ports
+        /// TFT Reset
```

Comment:
A few of these have part of the documentation as a comment and part as actual docs. Is there a reason for that?

Reply:
i carried that over from the previous version, from what i've been able to discern the comments are to indicate logical groupings of pins in the code, while the documentation is for to document the pin itself